inc/saidot1brfdb.h

```diff
@@ -0,0 +1,130 @@
+/*
+* Copyright (c) 2015 Dell Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may
+* not use this file except in compliance with the License. You may obtain
+* a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*
+* THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR
+* CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+* LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+* FOR A PARTICULAR PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+*
+* See the Apache Version 2.0 License for specific language governing
+* permissions and limitations under the License.
+*
+* @file saidot1brfdb.h
+*
+* @brief This module defines SAI API for IEEE 802.1BR ECID based forwarding
+         functionality. This is applicable only to Port Extenders (PE).
+*************************************************************************************/
+
+#if !defined (__SAIDOT1BRFDB_H)
+#define __SAIDOT1BRFDB_H
+
+#include "saitypes.h"
+#include "saistatus.h"
+
+/** \defgroup SAIDOT1BRFDBENTRY SAI - 802.1BR ECID based forwarding specific public APIs and datastructures in PE.
+ *
+ *  \{
+ */
+
+/**
+ * @brief SAI attributes for SAI_OBJECT_TYPE_DOT1BR_FDB_ENTRY 
+ */
+typedef enum _sai_dot1br_fdb_entry_attr_t
```

Comment:
there is plan to convert fdb_entry_t to actual sai_object_id_t and mac and vlan could mandatory attributes, Type attribute could be introduced as mandatory attribute to determine wheter its fdb entry or fdb_dot1br entry and based on type only specific attributes would apply to specific fdb entry type.
IMO we dont need more unions, they make things more complicated and next to union we still need a type to know which union member is used. it would be better to use attributes instead